frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/vms/NewVmModelBehavior.java

```diff
@@ -205,6 +206,16 @@ private void selectedTemplateChanged(final VmTemplate template) {
             isoPath = template.getIsoPath();
             // Random Generator
             updateRngDevice(template.getId());
+            // VirtIO SCSI
+            int queues = template.getVirtioScsiMultiQueues();
+            if (queues == -1) {
+                getModel().getVirtioScsiMultiQueueTypeSelection().setSelectedItem(VirtioMultiQueueType.AUTOMATIC);
+            } else if (queues == 0) {
+                getModel().getVirtioScsiMultiQueueTypeSelection().setSelectedItem(VirtioMultiQueueType.DISABLED);
+            } else {
+                getModel().getVirtioScsiMultiQueueTypeSelection().setSelectedItem(VirtioMultiQueueType.CUSTOM);
+                getModel().getNumOfVirtioScsiMultiQueues().setEntity(queues);
+            }
```

Comment:
how about extracting this to a new method `updateVirtioScsiMultiQueue` that can also be used from the other behavior class?

Reply:
The `virtioScsiMultiQueues` is copied in HwOnlyVmBaseToUnitBuilder. However, that builder is not used in `NewVmModelBehavior`. The "hardware" data are copied in the `InstanceTypeManager`. To be consistent, it would be best to set the `virtioScsiMultiQueues` there too. As the pools use the instance type manager too it would be worth testing if that solves also the issue with pools.

Reply:
@smelamud so let's do that?

Reply:
Done.